gcsfs/core.py

```diff
@@ -835,18 +835,6 @@ async def _info(self, path, generation=None, **kwargs):
         else:
             raise FileNotFoundError(path)
 
-    async def _glob(self, path, prefix="", **kwargs):
-        if not prefix:
-            # Identify pattern prefixes. Ripped from fsspec.spec.AbstractFileSystem.glob and matches
-            # the glob.has_magic patterns.
-            indstar = path.find("*") if path.find("*") >= 0 else len(path)
-            indques = path.find("?") if path.find("?") >= 0 else len(path)
-            indbrace = path.find("[") if path.find("[") >= 0 else len(path)
-
-            ind = min(indstar, indques, indbrace)
-            prefix = path[:ind].split("/")[-1]
-        return await super()._glob(path, prefix=prefix, **kwargs)
```

Comment:
If we apply the fix in the `_find` method below then this glob prefix construction no longer work for the edge case highlighted [here](https://github.com/fsspec/filesystem_spec/pull/1340/files#diff-1b7970a1cf6d8e169b2eab44ba722468be130bfeeef9deaed64e0d89882b88edR516).

This is because the the glob `subdir*` will create a `subdir` prefix and will be handled like a folder in the `_find` method (and add a trailing slash and miss the `subdir.txt` file).

Also, not sure to understand the initial benefit of this `_glob` method override. Was this just for caching?
Anyway, if another fix exists maybe this could be left as-is.